File: PracticeWork04CS/Animals.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PracticeWork04CS
{
    abstract class IAnimals
    {
        public abstract void Speak();
    }
    class Cat : IAnimals
    {
        public Cat()
        {
            Count++;
        }
        public override void Speak()
        {
            Console.WriteLine("Meow");
        }
        static int Count = 0;
    }
    class Dog : IAnimals
    {
        public Dog()
        {
            Count++;
        }
        public override void Speak()
        {
            Console.WriteLine("Haw");
        }
        static int Count = 0;
    }
    class Humster : IAnimals
    {
        public Humster()
        {
            Count++;
        }
        public override void Speak()
        {
            Console.WriteLine("Hmff");
        }
        static int Count = 0;
    }
    class Hippo : IAnimals
    {
        public Hippo()
        {
            Count++;
        }
        public override void Speak()
        {
            Console.WriteLine("AAAA");
        }
        static int Count = 0;
    }
}

File: PracticeWork04/Source/private/Animals.cpp
#include "../public/Animals.h"
#include <iostream>

int Cat::Count = 0;
int Dog::Count = 0;
int Humster::Count = 0;
int Hippo::Count = 0;


void Cat::Speak()
{
	std::cout << "Meow" << std::endl;
}

void Dog::Speak()
{
	std::cout << "Haw" << std::endl;
}

void Humster::Speak()
{
	std::cout << "Hmff" << std::endl;
}

void Hippo::Speak()
{
	std::cout << "AAAAA" << std::endl;
}
